src/parsers/psse_dynamic_data.jl

```diff
@@ -0,0 +1,267 @@
+        else
+            _t = strip(_v, ['(', ')'])
+            _t2int = parse.(Int64, split(_t, ','))
+            struct_args[ix] = (val[_t2int[1]], val[_t2int[2]])
+        end
+    end
+    return struct_args
+end
+
+"""
+Create a SingleMass shaft struct directly using the parameter mapping.
+
+"""
+function _make_shaft(param_map, val)
+    constructor_shaft =
+        (args...) -> InteractiveUtils.getfield(PowerSystems, :SingleMass)(args...)
+    shaft_args = _populate_args(param_map, val)
+    return constructor_shaft(shaft_args...)
+end
+
+function _make_source(g::StaticInjection, r::Float64, x::Float64)
+    return Source(
+        name = get_name(g),
+        available = true,
+        bus = get_bus(g),
+        active_power = get_active_power(g),
+        reactive_power = get_reactive_power(g),
+        R_th = r,
+        X_th = x,
+    )
+end
+
+function _assign_missing_components!(bus_dict, component_table)
+    for va in values(bus_dict)
+        !isassigned(va, component_table["AVR"]) &&
+            (va[component_table["AVR"]] = AVRFixed(1.0))
+        !isassigned(va, component_table["TurbineGov"]) &&
+            (va[component_table["TurbineGov"]] = TGFixed(1.0))
+        !isassigned(va, component_table["PSS"]) &&
+            (va[component_table["PSS"]] = PSSFixed(0.0))
+    end
+end
+
+"""
+Parse a .dyr file directly from its name by constructing its dictionary of dictionaries.
+
+"""
+function _parse_dyr_components(dyr_file::AbstractString)
+    ext = splitext(dyr_file)[2]
+    if lowercase(ext) in [".dyr"]
+        data = _parse_dyr_file(dyr_file)
+        return _parse_dyr_components(data)
+    else
+        throw(DataFormatError("$dyr_file is not a .dyr file type"))
+    end
+end
+
+"""
+Parse dictionary of dictionaries of data (from `_parse_dyr_file`) into a dictionary of struct components.
+The function receives the parsed dictionary and construct a dictionary indexed by bus, that containts a 
+dictionary with each dynamic generator components (indexed via its id).
+
+Each dictionary indexed by id contains a vector with 5 of its components:
+* Machine
+* Shaft
+* AVR
+* TurbineGov
+* PSS
+
+"""
+function _parse_dyr_components(data::Dict)
+    yaml_mapping = open(PSSE_DYR_MAPPING_FILE) do io
+        aux = YAML.load(io)
+        return aux
+    end
+    #param_map contains the mapping between struct params and dyr file.
+    param_map = yaml_mapping["parameter_mapping"][1]
+    #dic will contain the dictionary index by bus.
+    #Each entry will be a dictionary, with id as keys, that contains the vector of components
+    dic = Dict{Int64, Any}()
+    component_table =
+        Dict("Machine" => 1, "Shaft" => 2, "AVR" => 3, "TurbineGov" => 4, "PSS" => 5)
+    for (bus_num, bus_data) in data
+        bus_dict = Dict{Int64, Any}()
+        for (componentID, componentValues) in bus_data
+            #Fill array of 5 components per generator
+            temp = get!(bus_dict, componentID[2], Vector{Any}(undef, 5))
```

Comment:
What happens with `temp`?  It doesn't appear to be used for anything.

Reply:
`temp` is the value that is stored in `bus_dict` with key `componentID[2]`. Basically, modifying `temp`, modifies the vector stored in `bus_dict`. The nice thing also, is that I can create the vector if that key is not defined yet.